src/mock-doc/node.ts

```diff
@@ -261,9 +266,11 @@ export class MockElement extends MockNode {
     }
   }
 
-  get attributes() {
+  get attributes(): MockAttributeMap {
     if (this.__attributeMap == null) {
-      this.__attributeMap = createAttributeProxy(false);
+      const attrMap = createAttributeProxy(false);
+      this.__attributeMap = attrMap;
+      return attrMap;
     }
     return this.__attributeMap;
   }
```

Comment:
this is a funny little bit here. I changed the type of `this.__attributeMap` to be `MockAttributeMap | null | undefined` which, afaict, reflects the types that the variable actually is set to by our code. This however causes the return value for this getter to be that type as well since the original implementation just returned `this.__attributeMap`:

https://github.com/ionic-team/stencil/blob/7f8e5c68a786cd82f8a60aaf4f0ee943ea6d5735/src/mock-doc/node.ts#L264-L269

Anywhoo - I wanted to make the type signature `() => MockAttributeMap` instead. When we check `this.__attributeMap == null` that narrows the type outside of the `if { }` block as long as we have a `return` inside of it, but unfortunately doing a reassignment like

```ts
this.__attributeMap = createAttributeProxy(false)
```

doesn't change the type of `this.__attributeMap`. So anyway, as a workaround I used a temporary variable `attrMap` (which will get the right type) and then just returned a reference to the same object that we stick on `this.__attributeMap` by returning that.